pyodide-build/pyodide_build/install_xbuildenv.py

```diff
@@ -96,6 +96,15 @@ def install(path: Path, *, download: bool = False, url: str | None = None) -> No
     from . import __version__
 
     version = __version__
+
     if download:
-        download_xbuildenv(version, path, url=url)
+        if path.exists():
+            logger.warning("xbuild environment already exists, skipping download")
+        else:
+            _download_xbuildenv(version, path, url=url)
+    else:
+        if not path.exists():
+            logger.error("xbuild environment not exists")
+            raise FileNotFoundError(path)
+
```

Comment:
Nitpick maybe this is slightly neater:
```suggestion
    if not download and not path.exists():
        logger.error("xbuild environment not exists")
        raise FileNotFoundError(path)

    if download and path.exists():
        logger.warning("xbuild environment already exists, skipping download")
    elif download:
        _download_xbuildenv(version, path, url=url)
```